regulations/generator/api_reader.py

```diff
@@ -75,11 +75,16 @@ def _get(self, cache_key_elements, api_suffix, api_params={}):
             self.cache.set(cache_key, element)
             return element
 
-    def layer(self, layer_name, label, version):
-        regulation_part = label.split('-')[0]
+    def layer(self, layer_name, doc_type, doc_id):
+        """When retrieving layer data, we cheat a bit -- we always retrieve
+        layer data corresponding to the "root" of the document, rather than
+        only a subnode"""
+        doc_id_parts = doc_id.split('/')
+        doc_id_parts[-1] = doc_id_parts[-1].split('-')[0]   # root
```

Comment:
This is a bit of a head-scratcher if you don't know that `doc_id` is either `"${label}" or "${version}/${label}"`. It might be worth showing work a bit more:

```python
label = doc_id.split('/')[-1]
root = label.split('-')[0]
```

We could even introduce a `namedtuple` or class that represents documents, which would handle versions and labels, know how to find the root of a label, etc. Likely not worthwhile, but there's a certain amount of cognitive load involved in this kind of string munging that we could avoid.